code/modules/projectiles/projectile/bullets/revolver.dm

```diff
@@ -118,3 +118,18 @@
 	ricochet_auto_aim_range = 6
 	ricochet_incidence_leeway = 80
 	ricochet_decay_chance = 1
+
+// used only by gatfruits, does moderate brute damage and slight toxin damage.
+/obj/projectile/bullet/pea
+	name = "pea bullet"
+	damage = 15
+	weak_against_armour = TRUE
+	ricochets_max = 2
+	ricochet_chance = 100
+	icon_state = "pea"
+
+/obj/projectile/bullet/pea/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(isliving(target))
```

Comment:
```suggestion
	if(. == BULLET_ACT_HIT && isliving(target))
```
If you only want the toxin damage to be dealt on a successful hit

Reply:
in the context of the bullet being blocked by something like a desword, or bouncing off armor?